### PR TITLE
Update Spaces docs for v1.0.1

### PR DESCRIPTION
**Pull request by branden**
This bumps version references and adds release notes.

### CODE REVIEW

File: content/reference/rel-notes.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ weight: 150
 Find below the release notes for all released versions of [Upbound Spaces]({{< ref "spaces/overview.md" >}}).
 
 <!-- vale off -->
+## Spaces v1.0.1 release notes
+
+- [Backport release-1.0] Export mxp-gateway metrics via otlp-collector by @github-actions in #332
```
Comment (tnthornton) [inline content/reference/rel-notes.md:11]: Could we make this external consumer friendly? Maybe remove the backport tag and the github action reference?

Comment (branden) [reply content/reference/rel-notes.md:11]: Sure, I'll clean up the 1.0.0 release notes too.

**Review by tnthornton — APPROVED** [review]

**Comment by AlainRoy** [comment]
Craig asked us to merge the Spaces docs on Wednesday:
https://github.com/upbound/docs/pull/228

Should I assume that we'll also need to push your change on Monday as well? @branden?

**Comment by branden** [comment]
@AlainRoy #228 is getting updated automatically as we merge to spaces-staging, so my changes are already there. They'll be included when we merge it.

**Pull request merged**